### PR TITLE
Dependency cleaning

### PR DESCRIPTION
**Pull request by wdanilo**
### Cleaning of the repo

- Obsolete `lib/rust/logger` was removed.
- Following prelude dependencies were removed (not only from prelude, from the whole project):
   - `cfg-if`
   - `colored`
   - `enclose`
   - `ifmt`
   - `num`
   - `object`
   - `shrinkwraprs`
- Many dependencies in `Cargo.lock` files were unified with the workspace versions.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.

### CODE REVIEW

File: app/gui/language/parser/Cargo.toml [hunk 1]
```diff
@@ -15,8 +15,8 @@ enso-prelude = { path = "../../../../lib/rust/prelude" }
 enso-profiler = { path = "../../../../lib/rust/profiler" }
 enso-text = { path = "../../../../lib/rust/text" }
 console_error_panic_hook = { version = "0.1.6" }
-failure = { version = "0.1" }
-js-sys = { version = "0.3.28" }
+failure = { workspace = true }
+js-sys = { workspace = true }
 matches = { version = "0.1" }
 serde = { version = "1.0", features = ["derive"] }
 serde_json = { version = "1.0", features = ["unbounded_depth"] }
```
Comment (farmaazon) [inline app/gui/language/parser/Cargo.toml:22]: I've seen serde_json pointing to the workspace. Also, we may move serde and uuid there. And this crate still uses shrinkwraps.

File: app/gui/view/debug_scene/text-grid-visualization/Cargo.toml [hunk 2]
```diff
@@ -13,9 +13,9 @@ ensogl = { path = "../../../../../lib/rust/ensogl" }
...
 ide-view = { path = "../.." }
-js-sys = { version = "0.3.28" }
-nalgebra = { version = "0.26.1" }
-serde_json = { version = "1.0" }
+js-sys = { workspace = true }
+nalgebra = { workspace = true }
+serde_json = { workspace = true }
 wasm-bindgen = { workspace = true }
 ensogl-list-view = { path = "../../../../../lib/rust/ensogl/component/list-view" }
 ensogl-grid-view = { path = "../../../../../lib/rust/ensogl/component/grid-view" }
```
Comment (farmaazon) [inline app/gui/view/debug_scene/text-grid-visualization/Cargo.toml:21]: web-sys could be defined in workspace as well.

File: app/gui/view/graph-editor/src/component/visualization/container/fullscreen.rs [hunk 3]
```diff
@@ -52,16 +52,13 @@ pub mod background {
 #[derive(Clone, CloneRef, Debug)]
 #[allow(missing_docs)]
 pub struct Panel {
-    logger:             Logger,
     display_object:     display::object::Instance,
-    // background     : background::View,
```
Comment (farmaazon) [inline app/gui/view/graph-editor/src/component/visualization/container/fullscreen.rs:57]: I think this field was part of the TODO in line 24.

File: lib/rust/prelude/src/lib.rs [hunk 4]
```diff
@@ -162,6 +158,21 @@ pub use enso_logging::trace_span;
 pub use enso_logging::warn;
 pub use enso_logging::warn_span;
 
+// FIXME: finish it
```
Comment (farmaazon) [inline lib/rust/prelude/src/lib.rs]: FIXME left

File: lib/rust/profiler/Cargo.toml [hunk 5]
```diff
@@ -5,7 +5,7 @@ edition = "2021"
 authors = ["Enso Team <contact@enso.org>"]
 
 [dependencies]
-futures = "0.3"
+futures = { workspace = true }
 serde = { version = "1.0", features = ["derive"] }
 serde_json = { version = "1.0.59", features = ["raw_value"] }
```
Comment (farmaazon) [inline lib/rust/profiler/Cargo.toml:10]: serde-json still not referring to the workspace.

File: lib/rust/shapely/Cargo.toml [hunk 6]
```diff
@@ -20,8 +20,8 @@ default = []
 
 [dependencies]
 enso-shapely-macros = { version = "^0.2.1", path = "macros" }
-paste = { version = "0.1" }
-derivative = { version = "2.2.0" }
+paste = { workspace = true }
+derivative = { workspace = true }
 shrinkwraprs = { version = "0.3.0" }
```
Comment (farmaazon) [inline lib/rust/shapely/Cargo.toml]: shrintwraprs still present

File: lib/rust/shapely/Cargo.toml [hunk 7]
```diff
@@ -20,8 +20,8 @@ default = []
 
 [dependencies]
 enso-shapely-macros = { version = "^0.2.1", path = "macros" }
```
Comment (farmaazon) [inline lib/rust/shapely/Cargo.toml]: version and path. I've seen you removed versions in such lines.

File: Cargo.toml [hunk 8]
```diff
@@ -84,3 +84,23 @@ tokio = { version = "1.23.0", features = ["full", "tracing"] }
 tokio-util = { version = "0.7.4", features = ["full"] }
 wasm-bindgen = { version = "0.2.83", features = ["serde-serialize"] }
 wasm-bindgen-test = { version = "0.3.33" }
```
Comment (farmaazon) [inline Cargo.toml:91]: I've seen wasm-bindgen-test not refering to workspace somewhere.

File: lib/rust/json-rpc/Cargo.toml [hunk 9]
```diff
@@ -14,7 +14,7 @@ enso-web = { path = "../web" }
 enso-profiler-data = { path = "../profiler/data" }
 enso-profiler = { path = "../profiler" }
 futures = { version = "0.3.1" }
-failure = { version = "0.1.6" }
+failure = { workspace = true }
 serde = { version = "1.0.0", features = ["derive"] }
-serde_json = { version = "1.0.0" }
+serde_json = { workspace = true }
 shrinkwraprs = { version = "0.3.0" }
```
Comment (farmaazon) [inline lib/rust/json-rpc/Cargo.toml]: shrinkwraprs still present.

File: lib/rust/ensogl/core/Cargo.toml [hunk 10]
```diff
@@ -31,18 +30,18 @@ bit_field = { version = "0.10.0" }
...
-nalgebra = { version = "0.26.1" }
+itertools = { workspace = true }
+js-sys = { workspace = true }
+nalgebra = { workspace = true }
 num_enum = { version = "0.5.1" }
 num-traits = { version = "0.2" }
 rustc-hash = { version = "1.0.1" }
 semver = { version = "1.0.9" }
 serde = { version = "1" }
 shrinkwraprs = { version = "0.3.0" }
```
Comment (farmaazon) [inline lib/rust/ensogl/core/Cargo.toml]: Still shrinkwraprs

**Review by farmaazon — APPROVED** [review]
Please fix the fixme in prelude. And remove all shrinkwraprs from Cargo.tomls, so we actually get rid of it.

Otherwise other comments are not obligatory (I'm not sure of desired scope of this PR).

File: app/gui/language/parser/Cargo.toml (hunk 1 above)
Comment (wdanilo) [reply app/gui/language/parser/Cargo.toml:22]: We want to move EVERYTHING there. It is just a lot of work and I wanted to do SOME cleaning here :) We will be progressing with it over time :)

File: app/gui/view/debug_scene/text-grid-visualization/Cargo.toml (hunk 2 above)
Comment (wdanilo) [reply app/gui/view/debug_scene/text-grid-visualization/Cargo.toml:21]: We want to move EVERYTHING there. It is just a lot of work and I wanted to do SOME cleaning here :) We will be progressing with it over time :)

**Comment by wdanilo** [comment]
All fixed.

**Pull request merged**